core/rpc-client/src/direct_client.rs

```diff
@@ -66,7 +66,7 @@ impl DirectApi for DirectClient {
 		let (port_in, port_out) = channel();
 
 		info!("[WorkerApi Direct]: (get) Sending request: {:?}", request);
-		WsClient::connect_one_shot(&self.url, request, &port_in)?;
+		WsClient::connect_one_shot(&self.url, request, port_in)?;
 		port_out.recv().map_err(Error::MspcReceiver)
```

Comment:
So this meant the connection remained open and for every call we made a new connection? 
Glad you found this, and the fix is very simple and elegant 👍

Reply:
I don't think the connection would stay open. Usually it would go like this:
- Received response is sent into the `Sender` inside the function
- 'recv()' returns
- Second `Sender` gets dropped at the end of the function

Reply:
Hm then I don't understand how this broke block production when one of the worker crashes?